python/ray/dag/compiled_dag_node.py

```diff
@@ -900,7 +900,7 @@ def _get_or_compile(
 
         from ray.dag.constants import RAY_ADAG_ENABLE_DETECT_DEADLOCK
 
-        if RAY_ADAG_ENABLE_DETECT_DEADLOCK and not self._detect_deadlock():
```

Comment:
The API behavior is 100% swapped without name change. Was the previous name wrong?

Reply:
> Was the previous name wrong?

Yes